geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/RecordUpdateProcessor.scala

```diff
@@ -39,6 +39,7 @@ import scala.util.control.NonFatal
  * 'modify' mode (which require the entire feature to be input), this processor only requires specifying
  * the fields that will be updated
  */
+@SupportsBatching
```

Comment:
We don't have any concerns about when the JavaDoc says not to use this annotation, right?

https://www.javadoc.io/doc/org.apache.nifi/nifi-api/latest/org/apache/nifi/annotation/behavior/SupportsBatching.html

Reply:
ah, i thought it would just give you multiple flow files at once :P
this was added by Hulbert back in the original processor and we've just been carrying it forward since. i'll check how we use it, the javadocs are a bit confusing. i don't think we ever call ProcessSession.commit, maybe we should be?

Reply:
> All methods that are called on a ProcessSession happen as a transaction. When we decided to end the transaction, we can do so either by calling commit() or by calling rollback(). Typically, this is handled by the AbstractProcessor class: if the onTrigger method throws an Exception, the AbstractProcessor will catch the Exception, call session.rollback(), and then re-throw the Exception. Otherwise, the AbstractProcessor will call commit() on the ProcessSession.

Reply:
we don't support transactions, so there's currently no way to ensure that an entire flow file is processed atomically. as such, we may end up writing or updating multiple times if for some reason a flow file fails part way through and is replayed. adding SupportsBatching would seem to make the problem "worse" in a sense, in that there could potentially be more files replayed on a failure. but it might also make the processing faster, particularly for many small files. and it seems like our ingests/updates need to be idempotent anyway. what do you think?

Reply:
> It is also important to note how this behavior is affected by using the org.apache.nifi.annotation.behavior.SupportsBatching annotation. If a Processor utilizes this annotation, calls to ProcessSession.commit may not take effect immediately. Rather, these commits may be batched together in order to provide higher throughput. However, if at any point, the Processor rolls back the ProcessSession, all changes since the last call to commit will be discarded and all "batched" commits will take effect. These "batched" commits are not rolled back.

Reply:
I'm fine continuing to carry the annotation forward.  Good notes about the annotation.